From 31aaa452a25088e664bad01c91be8fd7a4e5cb65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Mon, 18 Mar 2019 10:48:54 +0200
Subject: [PATCH 102/235] crypto: caam - fix compilation error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix an issue introduced by backporting the commit mentined in Fixes tag:

drivers/crypto/caam/caamalg_qi.c: In function ‘caam_qi_algapi_init’:
drivers/crypto/caam/caamalg_qi.c:3286:17: error: ‘struct caam_drv_private’ has no member named ‘qidev’
    dev_warn(priv->qidev, "%s alg allocation failed\n",
                 ^~

Fixes: 27c1c2f9b5e5 ("crypto: caam/qi - don't allocate an extra platform device")
Signed-off-by: Horia Geantă <horia.geanta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg_qi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/caamalg_qi.c b/drivers/crypto/caam/caamalg_qi.c
index 1e47406de26d..37f2b4a4e0b7 100644
--- a/drivers/crypto/caam/caamalg_qi.c
+++ b/drivers/crypto/caam/caamalg_qi.c
@@ -3021,6 +3021,7 @@ static int caam_init_common(struct caam_ctx *ctx, struct caam_alg_entry *caam,
 			    bool uses_dkp)
 {
 	struct caam_drv_private *priv;
+	struct device *dev;
 	/* Digest sizes for MD5, SHA1, SHA-224, SHA-256, SHA-384, SHA-512 */
 	static const u8 digest_size[] = {
 		MD5_DIGEST_SIZE,
@@ -3277,7 +3278,7 @@ int caam_qi_algapi_init(struct device *ctrldev)
 		t_alg = caam_alg_alloc(alg);
 		if (IS_ERR(t_alg)) {
 			err = PTR_ERR(t_alg);
-			dev_warn(priv->qidev, "%s alg allocation failed\n",
+			dev_warn(ctrldev, "%s alg allocation failed\n",
 				 alg->driver_name);
 			continue;
 		}
-- 
2.17.1

